### DISCUSSION TITLE
How to programmatically unban user?

### DISCUSSION THREAD

**Discussion by ghost** [nicotine-plus/nicotine-plus#2226] [Q&A] [answered]
How do I undo this?
```
self.core.network_filter.ban_user(user)
self.core.network_filter.block_user_ip(user)
self.probed[user] = 'processed'
```
This is from the ban function in a modified Leech Detector plugin.

**Comment by slook** [comment, marked as answer]
`self.core.network_filter.unban_user(user)`
`self.core.network_filter.unblock_user_ip(user)`

https://github.com/nicotine-plus/nicotine-plus/blob/ff9c6c43f0de29e6fa329352d2d9e810bff0e182/pynicotine/networkfilter.py#L219-L244
